### ISSUE TITLE
Distribute Syncthing through Microsoft Store

### ISSUE THREAD

**Issue by joernheissler** [syncthing/syncthing#8178]
Hello,

I asked a Windows user to install Syncthing. Their first three comments were "there is no installer", "it's not signed" and "it's not in the Microsoft Store".

Would it be feasible to distribute Syncthing through the [Microsoft Store](https://en.wikipedia.org/wiki/Microsoft_Store) to improve Windows User experience a lot?

**Comment by bt90** [comment]
Signing should work again soon™


#8072

**Comment by AudriusButkevicius** [comment]
It's not a UWP app so not sure it would fit well.

I think our preferred way of distribution is "just download it from github".

We don't need an installer as there is nothing to install, just run it.

Granted, it's somewhat grim as you have to do things to get it to launch on startup, but I don't see how being in a UWP sandbox makes that experience any better, other adding maintenance pain for us (when was dealing with appstores easy?)

Also, who even uses MS app store? The experience I had with is is horrible.

**Comment by tomasz1986** [comment]
Maybe not Syncthing itself, but it'd probably be useful if SyncTrayzor was available in the Store. It's already the recommended way to install Syncthing on Windows for new users / beginners, so it'd likely fit much better there than the standard command-line binary.

**Comment by imsodin** [comment]
Personal opinion: I'd rather not do this under the Syncthing project umbrella. Not because of my dislike of Microsoft, but because of what Audrius said:

> when was dealing with appstores easy?

Once it's there, we are expected to keep it there and working.

I would agree to "sanction" someone from the community providing Syncthing in the Microsoft store ("sanction" because I think we do have control over the name/trademark, not related to the source license).

**Comment by NomadicVoxel** [comment]
> It's not a UWP app so not sure it would fit well.

I agree that it would be more elegant if there was a version that matched the UWP style, but I don't think that's a requirement at all. Many other non-UWP software are available in Microsoft's appstore, opensource or not. Inkscape is one example of an app that... could use some UX work, but ships through the appstore too. 

> Also, who even uses MS app store? The experience I had with is is horrible.

- Since when are more options a bad thing? Sure, you don't like it, but some do. 
- It would be beneficial to what users do use the store for the dev team to fill that niche before someone else does. *Somebody* is going to clone this and put in ads, look at what happened to GIMP. 
- Windows S mode is a thing, which only permits software from the appstore. Tangent, but while power users don't like it, it's useful for contexts where you want a less tech-savvy user to stop downloading strange software from who knows where. (Not the best solution, but it's something.) 
- The only way their market is going to get better is if people start publishing *good* things and flagging the bad. Of course, Microsoft could help a bit, but all software markets with minimal quality control look like this.

**Comment by bt90** [comment]
Right after they stepped back from a controversial policy change regarding open source software?

https://docs.microsoft.com/en-us/windows/uwp/publish/store-policies-change-history

> Update to 10.8.7 to prohibit charging fees in the Store for open-source or other software that is generally available for free [...]

They tried to clean up their mess of a store and were fine with throwing valid monetization strategies of projects like Krita under the bus for it.

After a valid shitstorm they delayed the changes (for now):

https://mobile.twitter.com/gisardo/status/1545473811226705922

The Software Freedom Conservancy published a blog post on the matter:

https://sfconservancy.org/blog/2022/jul/07/microsoft-bans-commerical-open-source-in-app-store/

As long as we can dodge the hell hole of a "managed" appstore, we should. The situation on Android is bad enough. I think the syncthing project had its fair share of playstore trouble to judge this. We shouldn't aim to support a similar abomination on Windows.

**Comment by NomadicVoxel** [comment]
Fair. I didn't hear about the issue with Krita. I do hope the mess blows over. 

And yeah Android is weird :\ Don't know much about the store from the dev's side, but given the restrictions on access to SD and other crap like that, I can tell it's not easy.

**Comment by lwbt** [comment]
Some input from my point of view and recent experience with Windows as a "I touch Windows only a few days a year on my personal devices" user, if I may: Many applications are now available through `winget`. Winget has different sources, the default source is `winget` and there is also `msstore`.

I think this is you starting point for: https://github.com/microsoft/winget-pkgs

I install a few applications like git, Neovim, jq, VS Code and others through a PowerShell script with winget and I'm done with my initial setup (and uninstalling/disabling components I don't need, like SMBv1).

I haven't published an application myself this way, but since so many are doing it and it supports a variety of formats from typical zip files to AppBundles and all the other formats... it may not be as involved and over engineered as AppBundles might appear and it might be the "right" thing for advanced users as well as users who are concerned about open source and freedom. Winget is part of the AppInstaller package, which should be installed and updated automatically, if not running this command on a Windows machine where the Microsoft Store is not locked down by an administrator gets you to the right page immediately:

```pwsh
Start-Process -FilePath ms-windows-store://pdp/?ProductId=9NBLGGH4NNS1
```

/OT

> given the restrictions on access to SD and other crap like that

Hmm, not sure what you mean and this is the wrong thread/project, but from my point of view restrictions couldn't be strict enough. There are still apps around which create and thus pollute the directory with all kinds of folders and files so that it doesn't feel like my personal space for data but for untalented developers to just dump all their garbage.

**Comment by rasa** [comment]
@lwbt Syncthing (via SyncTrayzor) is already in winget: https://github.com/microsoft/winget-pkgs/blob/master/manifests/s/SyncTrayzor/SyncTrayzor/1.1.29.0/SyncTrayzor.SyncTrayzor.installer.yaml

**Comment by mborsetti** [comment]
See bug in #10676
